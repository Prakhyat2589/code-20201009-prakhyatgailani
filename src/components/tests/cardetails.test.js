import React from "react";
import { render, screen } from "@testing-library/react";
import CarDetails from "../cardetails";

const mockResponse = {
  carData: {
    Results: [
      {
        Make_ID: 1,
        Make_Name: "Car1",
      },
      {
        Make_ID: 2,
        Make_Name: "Car2",
      },
      {
        Make_ID: 3,
        Make_Name: "Car3",
      },
    ],
  },
};

const CarItemListComponent = (mockprops) => {
  const { key } = "carkey";
  return render(<CarDetails key={key} {...mockprops}></CarDetails>);
};

describe("Unit test for CarDetails", () => {
  test("renders CarDetails", async () => {
    const { container } = CarItemListComponent(mockResponse);
    expect(container).toMatchSnapshot();
  });
});
