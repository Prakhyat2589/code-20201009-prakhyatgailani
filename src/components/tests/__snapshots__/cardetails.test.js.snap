// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for CarDetails renders CarDetails 1`] = `
<div>
  <h1>
    Car Listings
  </h1>
  <div
    class="sc-bdfBQB bSCONV"
  >
    <div
      class="sc-gsTEea hFEVSK"
    >
      <div
        class="sc-dlfnuX sc-hKgJUU fNPEMh ifIJfe"
      >
        CAR ID
      </div>
      <div
        class="sc-dlfnuX sc-hKgJUU fNPEMh ifIJfe"
      >
        CAR NAME
      </div>
    </div>
    <div
      class="sc-gsTEea hFEVSK"
    >
      <div
        class="sc-dlfnuX fNPEMh"
      >
        1
      </div>
      <div
        class="sc-dlfnuX fNPEMh"
      >
        Car1
      </div>
    </div>
    <div
      class="sc-gsTEea hFEVSK"
    >
      <div
        class="sc-dlfnuX fNPEMh"
      >
        2
      </div>
      <div
        class="sc-dlfnuX fNPEMh"
      >
        Car2
      </div>
    </div>
    <div
      class="sc-gsTEea hFEVSK"
    >
      <div
        class="sc-dlfnuX fNPEMh"
      >
        3
      </div>
      <div
        class="sc-dlfnuX fNPEMh"
      >
        Car3
      </div>
    </div>
  </div>
</div>
`;
